wandb/sdk/internal/sender.py

```diff
@@ -991,12 +997,52 @@ def finish(self):
             self._fs = None
 
     def _max_cli_version(self):
-        _, server_info = self._api.viewer_server_info()
+        server_info = self.get_server_info()
         max_cli_version = server_info.get("cliVersionInfo", {}).get(
             "max_cli_version", None
         )
         return max_cli_version
 
+    def get_viewer_server_info(self):
+        if not self._cached_server_info or not self._cached_viewer:
+            (
+                self._cached_viewer,
+                self._cached_server_info,
+            ) = self._api.viewer_server_info()
+
+    def get_viewer_info(self):
+        if not self._cached_viewer:
+            self.get_viewer_server_info()
+        return self._cached_viewer
+
+    def get_server_info(self):
+        if not self._cached_server_info:
+            self.get_viewer_server_info()
+        return self._cached_server_info
+
+    def get_local_info(self):
+        """
+        This is a helper function that queries the server to get the the local version information.
+        First, we perform an introspection, if it returns empty we deduce that the docker image is
+        out-of-date. Otherwise, we use the returned values to deduce the state of the local server.
+        """
+        local_info = wandb_internal_pb2.LocalInfo()
+
+        latest_local_version = "latest"
+
+        # Assuming the query is succesful if the result is empty it indicates that
+        # the backend is out of date since it doesn't have the desired field
+        server_info = self.get_server_info()
+        latest_local_version_info = server_info.get("latestLocalVersionInfo", {})
+        if latest_local_version_info is None:
```

Comment:
One case that I already observed is that before the jobber runs and populates all the fields it will be `None`. The jobber suppose to run every midnight (or 24 hours) as far as i understand, after that we should have valid values.

To recreate this you can spin up a local docker and and try to query it, you will get `None` unless you explicitly run the script the jobber runs, or wait for it to be updated

Reply:
Ah no problem :)